pkg/workloads/reservation.go

```diff
@@ -1318,6 +1295,84 @@ func (a *API) setFarmIPFree(db *mongo.Database, workload types.WorkloaderType) e
 	return nil
 }
 
+func (a *API) handleKubernetesReservation(ctx context.Context, db *mongo.Database, workload types.WorkloaderType, userID int64) mw.Response {
+	k8sWorkload := workload.Workloader.(*generated.K8S)
+
+	if k8sWorkload.PublicIP == 0 {
+		return nil
+	}
+
+	var err error
+
+	var workloadFiler types.WorkloadFilter
+	workloadFiler = workloadFiler.WithID(k8sWorkload.PublicIP)
+	ipWorkload, err := workloadFiler.Get(ctx, db)
+	if err != nil {
+		return mw.NotFound(errors.Wrapf(err, "ip workload %d not found", k8sWorkload.ID))
+	}
+
+	if ipWorkload.GetNextAction() != generated.NextActionDeploy {
```

Comment:
You know your both `next-action` and `customer-tid` can added to your workloadFilter. query and then do Get one time 
if u get Error (not found) then the user is not allowed to use this ip (for any reason that be)

Reply:
I know I can use a filter, but then I cannot throw a custom error for the 2 cases?